pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -266,7 +268,12 @@ func (h *NetworkUtilsHandler) GetNFTIPString(proto iptables.Protocol) string {
 	return "ip"
 }
 
-func (h *NetworkUtilsHandler) NftablesLoad(fnName string) error {
+func (h *NetworkUtilsHandler) NftablesLoad(proto iptables.Protocol) error {
+	ipVersion := "4"
+	if proto == iptables.ProtocolIPv6 {
+		ipVersion = "6"
+	}
+	fnName := fmt.Sprintf("ipv%s-nat", ipVersion)
```

Comment:
Now we can unit test this moving the logic that compose the function into a helper (it does not need a receiver) that returns`*exe.Cmd`, `NftablesLoad` will call it to compose the command and run it.

```golang
func composeNftablesLoad(proto iptables.Protocol) *exec.Cmd {
ipVersion := "4"
	if proto == iptables.ProtocolIPv6 {
		ipVersion = "6"
	}
	fnName := fmt.Sprintf("ipv%s-nat", ipVersion)
        return exec.Command("nft", "-f", fmt.Sprintf("/etc/nftables/%s.nft", fnName))
}

func (h *NetworkUtilsHandler) NftablesLoad(proto iptables.Protocol) error {
    output, err := composeNftablesLoad().CombinedOutput()
    ...
}

```

Reply:
I don't think that `composeNftablesLoad` needs a uit test.
It has no logic.
What we are testing? That `exec.Command` is working? (That's not our scope to test it).
That the `Sprintf` logic is working? It just inserts 4 or 6 to the file name. I don't think it deserves a unit test.

Reply:
> I don't think that `composeNftablesLoad` needs a uit test.
> It has no logic.
> What we are testing? That `exec.Command` is working? (That's not our scope to test it).

IIUC, it would be that we create the expected `exec.Command` commands.

> That the `Sprintf` logic is working? It just inserts 4 or 6 to the file name. I don't think it deserves a unit test.

Reply:
> I don't think that `composeNftablesLoad` needs a uit test.
> It has no logic.
> What we are testing? That `exec.Command` is working? (That's not our scope to test it).
> That the `Sprintf` logic is working? It just inserts 4 or 6 to the file name. I don't think it deserves a unit test.

We have to test that depending on protocl we compose correctly the nftable name, we are checking the content of exe.Cmd*.

Reply:
Although I believe it is an overkill to test this super simple method. Added a unit test. Done.